tests/test_io.py

```diff
@@ -156,3 +156,21 @@ def test_read_multi_frame_seg_image_sm_dots_bitpacked(self):
                     reader.metadata.Columns,
                 )
                 np.testing.assert_array_equal(frame, pixel_array[i, ...])
+
+    def test_read_single_frame_ct_image_dicom_file_like(self):
+        filename = str(self._test_dir.joinpath("ct_image.dcm"))
+        dcm = DicomBytesIO(open(filename, "rb").read())
+
+        dataset = dcmread(filename)
+        pixel_array = dataset.pixel_array
+        with ImageFileReader(dcm) as reader:
+            assert reader.number_of_frames == 1
+            frame = reader.read_frame(0)
+            assert isinstance(frame, np.ndarray)
+            assert frame.ndim == 2
+            assert frame.dtype == np.int16
+            assert frame.shape == (
+                reader.metadata.Rows,
+                reader.metadata.Columns,
+            )
+            np.testing.assert_array_equal(frame, pixel_array)
```

Comment:
We definitely need a test for the case where the input to the `ImageFileReader` is a `DicomFileLike` object that represents an opened file handle (and *not* a `DicomBytesIO`). I know this isn't really what you signed up for @RobinFrcd but I'd be more comfortable doing it in your branch so that we know everything is working before we merge. Therefore could you please accept this suggestion to add this test at the end of the file:

```suggestion
            np.testing.assert_array_equal(frame, pixel_array)
            
    def test_read_single_frame_ct_image_dicom_file_like_opened(self):
        # Test reading frames from an opened DicomFileLike file
        filename = self._test_dir.joinpath("ct_image.dcm")
        dcm = DicomFileLike(filename.open("rb"))

        dataset = dcmread(filename)
        pixel_array = dataset.pixel_array
        with ImageFileReader(dcm) as reader:
            assert reader.number_of_frames == 1
            frame = reader.read_frame(0)
            assert isinstance(frame, np.ndarray)
            assert frame.ndim == 2
            assert frame.dtype == np.int16
            assert frame.shape == (
                reader.metadata.Rows,
                reader.metadata.Columns,
            )
            np.testing.assert_array_equal(frame, pixel_array)
```

We should not merge until both new tests are passing